platform/platform-paxweb-jettyconfig/src/main/java/org/codice/ddf/security/session/AttributeSharingHashSessionIdManager.java

```diff
@@ -83,37 +78,26 @@
 
     @Override
     public void invalidateSession(
-        String subjectName, Function<HttpSession, String> sessionSubjectExtractor) {
+        String subjectName, Function<Map<String, Object>, String> sessionSubjectExtractor) {
 
-      final List<AbstractSession> matchingSessions =
-          getHttpSessionStream()
-              .filter(s -> subjectName.equals(sessionSubjectExtractor.apply(s)))
-              .filter(AbstractSession.class::isInstance)
-              .map(AbstractSession.class::cast)
-              .distinct()
-              .collect(Collectors.toList());
-      for (AbstractSession matchingSession : matchingSessions) {
-        idManager.removeSession(matchingSession);
-        if (matchingSession.isValid()) {
-          matchingSession.invalidate();
-        }
-      }
-    }
+      final Optional<String> sessionIdOptional =
+          getSessionAttributeMapsStream()
+              .filter(e -> subjectName.equals(sessionSubjectExtractor.apply(e.getValue())))
+              .map(e -> e.getKey())
+              .findFirst();
 
-    private Stream<HttpSession> getHttpSessionStream() {
-      return getFullSessions()
-          .stream()
-          .flatMap(Collection::stream)
-          .map(Reference::get)
-          .filter(Objects::nonNull);
+      if (sessionIdOptional.isPresent()) {
+        idManager.invalidateSession(sessionIdOptional.get());
```

Comment:
Good catch. I added that check in the `invalidateSession` method. I'm honestly not sure what scenario would cause an invalid session to be triggered to be invalidated, but better to be safe.